services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -38,11 +44,21 @@
              :path staging-dest})))
   data-item)
 
+(defn- validate-publish-dest
+  [{:keys [path] :as data-item}]
+  (let [publish-dest (ft/path-join (config/permanent-id-publish-dir) (ft/basename path))]
+    (when (data-info/path-exists? (config/irods-user) publish-dest)
+      (throw+ {:type :clojure-commons.exception/exists
+               :error "A folder with this name has already been published."
+               :path publish-dest})))
```

Comment:
This validation isn't throwing the exception only because we're trying to move the folder and the name in the destination already exists. For example, when the user makes the initial request, we validate that name doesn't exist in the `curated` folder, but it's only moving into the `staging` folder for now.

So this exception doesn't have to be `clojure-commons.exception/exists`. It could be a `bad-request` or even eventually become a `permanent-id-request` specific error code.